tests/h/traversal/roots_test.py

```diff
@@ -0,0 +1,197 @@
+        return service
+
+
+class TestAuthClientResourceFactory(object):
+    def test_get_item_returns_an_authclient(self, pyramid_request):
+        authclient = AuthClient(name='test', authority='example.com')
+        pyramid_request.db.add(authclient)
+        pyramid_request.db.flush()
+
+        factory = AuthClientRoot(pyramid_request)
+        assert factory[authclient.id] == authclient
+
+    def test_get_item_returns_keyerror_if_not_found(self, pyramid_request):
+        factory = AuthClientRoot(pyramid_request)
+        with pytest.raises(KeyError):
+            factory['E19D247D-1F07-4E91-B40D-00DF22E693E4']
+
+    def test_get_item_returns_keyerror_if_invalid(self, pyramid_request):
+        factory = AuthClientRoot(pyramid_request)
+        with pytest.raises(KeyError):
+            factory['not-a-uuid']
+
+
+@pytest.mark.usefixtures('organizations')
+class TestOrganizationRoot(object):
+
+    def test_it_returns_the_requested_organization(self, organizations, organization_factory):
+        organization = organizations[1]
+
+        assert organization_factory[organization.pubid] == organization
+
+    def test_it_404s_if_the_organization_doesnt_exist(self, organization_factory):
+        with pytest.raises(KeyError):
+            organization_factory['does_not_exist']
+
+    @pytest.fixture
+    def organization_factory(self, pyramid_request):
+        return OrganizationRoot(pyramid_request)
+
+
+@pytest.mark.usefixtures('organizations')
+class TestOrganizationLogoRoot(object):
+
+    def test_it_returns_the_requested_organizations_logo(self, organizations, organization_logo_factory):
+        organization = organizations[1]
+        organization.logo = '<svg>blah</svg>'
+
+        assert organization_logo_factory[organization.pubid] == '<svg>blah</svg>'
+
+    def test_it_404s_if_the_organization_doesnt_exist(self, organization_logo_factory):
+        with pytest.raises(KeyError):
+            organization_logo_factory['does_not_exist']
+
+    def test_it_404s_if_the_organization_has_no_logo(self, organizations, organization_logo_factory):
+        with pytest.raises(KeyError):
+            assert organization_logo_factory[organizations[0].pubid]
+
+    @pytest.fixture
+    def organization_logo_factory(self, pyramid_request):
+        return OrganizationLogoRoot(pyramid_request)
+
+
+@pytest.mark.usefixtures("groups")
+class TestGroupRoot(object):
+
+    def test_it_returns_the_group_if_it_exists(self, factories, group_factory):
+        group = factories.Group()
+
+        assert group_factory[group.pubid] == group
+
+    def test_it_raises_KeyError_if_the_group_doesnt_exist(self, group_factory):
+        with pytest.raises(KeyError):
+            group_factory["does_not_exist"]
+
+    @pytest.fixture
+    def groups(self, factories):
+        # Add some "noise" groups to the DB.
+        # These are groups that we _don't_ expect GroupRoot to return in
+        # the tests.
```

Comment:
Can you expand this comment to explain _why_ we'd want "noise" in the DB?

Reply:
This is pre-existing but I guess the reason is that it should return the _right_ group and the tests shouldn't pass if it e.g. just returns the first group in the DB or something